test/sol5/kyberNetworkProxy.js

```diff
@@ -578,6 +602,392 @@ contract('KyberNetworkProxy', function(accounts) {
+                srcQty, 
+                dstToken.address, 
+                destAddress,
+                maxDestAmt,
+                proxyRate.expectedRate,
+                platformWallet, 
+                emptyHint, { from: taker });
+            let dstQty = Helper.calcDstQty(srcQty, tokenDecimals[2], tokenDecimals[4], proxyRate.expectedRate);
+            let actualBalance = await dstToken.balanceOf(destAddress);
+            let actualDstQty = actualBalance.sub(initialTakerBalance);
+            await Helper.assertApproximate(dstQty, actualDstQty, "wrong balance");
+
+        });
+    });    
+
+    describe("test doTrade verify condition", async () => {
+        let generousNetwork;
+        let generousNetwork2;
+        let maliciousNetwork;
+        before("init 'generous' network and 'malicious' network", async () => {
+            // set up generousNetwork
+            generousNetwork = await GenerousNetwork.new(admin);
+            await nwHelper.setupNetwork(generousNetwork, networkProxy.address, KNC.address, DAO.address, tokens, accounts, admin, operator);
+            let result = await nwHelper.setupReserves(generousNetwork, tokens, 1, 1, 0, 0, accounts, admin, operator);
+            await nwHelper.addReservesToNetwork(generousNetwork, result.reserveInstances, tokens, operator);
+            // set up maliciousNetwork
+            maliciousNetwork = await MaliciousNetwork.new(admin);
+            await nwHelper.setupNetwork(maliciousNetwork, networkProxy.address, KNC.address, DAO.address, tokens, accounts, admin, operator);
+            result = await nwHelper.setupReserves(maliciousNetwork, tokens, 1, 1, 0, 0, accounts, admin, operator);
+            await nwHelper.addReservesToNetwork(maliciousNetwork, result.reserveInstances, tokens, operator);
+            // set up generousNetwork2
+            generousNetwork2 =await GenerousNetwork2.new(admin);
+            await nwHelper.setupNetwork(generousNetwork2, networkProxy.address, KNC.address, DAO.address, tokens, accounts, admin, operator);
+            result = await nwHelper.setupReserves(generousNetwork2, tokens, 1, 1, 0, 0, accounts, admin, operator);
+            await nwHelper.addReservesToNetwork(generousNetwork2, result.reserveInstances, tokens, operator);
+        });
+
+        it("trade revert if src address is not eth and msg value is not zero", async () => {
+            let srcToken = tokens[1];
+            let srcQty = new BN(3).mul(new BN(10).pow(new BN(tokenDecimals[1])));
+
+            //get rate
+            let rate = await networkProxy.getExpectedRate(srcToken.address, tokens[2].address, srcQty);
+            dstQty = Helper.calcDstQty(srcQty, tokenDecimals[1], tokenDecimals[2], rate.expectedRate);
+            await srcToken.transfer(taker, srcQty);
+            await srcToken.approve(networkProxy.address, srcQty, { from: taker });
+
+            //see trade reverts 
+            await expectRevert.unspecified(networkProxy.trade(
```

Comment:
must check the revert message

Reply:
updated in  https://github.com/KyberNetwork/smart-contracts/pull/726/commits/e4d067f71b9400d43a678940a586bbb43991f02a